test/helpers/gather-os-metrics.spec.js

```diff
@@ -0,0 +1,20 @@
+const chai = require('chai');
+
+chai.should();
+
+const gatherOsMetrics = require('../../helpers/gather-os-metrics');
+
+describe('helpers', () => {
+  describe('gather-os-metrics', () => {
+    describe('when invoked', () => {
+      it('then ...', () => {
+        const span = { os: [], responses: [] };
+
+        gatherOsMetrics({}, span);
+
+        // todo: not sure what should I test, maybe the resulted span structure?
+        // todo: also this component has got some internal timing events?
```

Comment:
For the pidusage error you'd have to test with a bad process id. Note that either you replace the global `process.pid` variable (not recommended) or you inject it in the function (I think it's cleaner). You may also mock pidusage.